From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Johannes Schindelin <johannes.schindelin@gmx.de>
Date: Thu, 30 Jul 2026 01:53:32 +0200
Subject: [PATCH 090/109] fixup! - Move root to /usr. - Change sorting mount
 points. - By default mount without ACLs. - Can read /etc/fstab with short
 mount point format.

(cherry picked from commit 126fb1d9750e87bdaa16eb6644401bb237a56ed2)
(cherry picked from commit 466b4746004957fd23e43ebc2839422ed2981ae2)
---
 winsup/testsuite/libltp/lib/search_path.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/winsup/testsuite/libltp/lib/search_path.c b/winsup/testsuite/libltp/lib/search_path.c
index df918287e..3e89fe3c0 100644
--- a/winsup/testsuite/libltp/lib/search_path.c
+++ b/winsup/testsuite/libltp/lib/search_path.c
@@ -139,8 +139,10 @@ printf("search_path: cmd = %s, access_mode = %d, fullpath = %d\n", cmd, access_m
 			PATH_MAX);
 		    return -1;
 		}
-		/* Don't emit "//cmd" when cwd is "/": a leading "//" is a UNC
-		   path on Cygwin/MSYS2.  */
+		/* Avoid a leading "//": on Cygwin/MSYS2 a path starting with
+		   "//" denotes a UNC network path.  When cwd is the root "/"
+		   (as in the MSYS2 testsuite install-root) "%s/%s" would yield
+		   "//cmd"; collapse curpath to "" so we emit "/cmd" instead. */
 		sprintf(res_path, "%s/%s", strcmp(curpath, "/") ? curpath : "", cmd);
 	    }
 	    else
@@ -204,7 +206,7 @@ printf("search_path: tmppath = %s\n", tmppath);
 	    continue;
 	}
 
-	/* Same "//" guard as above.  */
+	/* Same UNC-path guard as above for the PATH-scan branch. */
 	sprintf(res_path, "%s/%s", strcmp(tmppath, "/") ? tmppath : "", cmd);
 #if DEBUG
 printf("search_path: res_path = '%s'\n", res_path);
